UnchainedLauncher.Core/Services/Mods/Registry/ModRegistryFactory.cs

```diff
@@ -0,0 +1,46 @@
+using DiscriminatedUnions;
+using UnchainedLauncher.Core.Services.Mods.Registry.Downloader;
+using UnchainedLauncher.Core.Utilities;
+
+namespace UnchainedLauncher.Core.Services.Mods.Registry {
+    public class ModRegistryFactory {
+        public static readonly JsonFactory<ModRegistryMetadata, IModRegistry> Instance =
+            new JsonFactory<ModRegistryMetadata, IModRegistry>(ToClassType, ToJsonType);
```

Comment:
A JsonFactory consists of a json model (The metadata model) and the class type, and then functions to convert between them.  From that we get methods which load/export a class to/from json.